ui/app/src/components/UploadDialog/util.ts

```diff
@@ -62,3 +64,21 @@ export interface UploadDialogContainerProps extends UploadDialogProps {
   setPendingChanges?(pending: boolean): void;
   onMinimized?(): void;
 }
+
+export const getResponseError = (responseText: string, formatMessage: IntlShape['formatMessage']) => {
+  try {
+    const parsed = JSON.parse(responseText);
+    if (parsed.response) {
```

Comment:
@rart let's check this tomorrow, the write-content response can't be parsed and ends up in the catch, so we still don't have a clear error message when it fails to upload